File: backend/app/schemas/course.py
from datetime import datetime
from pydantic import BaseModel
from typing import List, Optional


class LessonSummary(BaseModel):
    id: int
    title: str
    slug: str
    lesson_type: str
    order: int
    duration_minutes: int
    is_completed: bool = False

    class Config:
        from_attributes = True


class ModuleWithLessons(BaseModel):
    id: int
    title: str
    description: Optional[str] = None
    order: int
    lessons: List[LessonSummary] = []

    class Config:
        from_attributes = True


class InstructorSummary(BaseModel):
    id: int
    full_name: str
    role: str
    bio: Optional[str] = None
    avatar_url: Optional[str] = None

    class Config:
        from_attributes = True


class CourseListItem(BaseModel):
    id: int
    title: str
    slug: str
    short_description: Optional[str] = None
    level: str
    estimated_hours: float
    modules_count: int
    lessons_count: int
    is_enrolled: bool = False
    progress_percentage: float = 0.0

    class Config:
        from_attributes = True


class CourseDetailResponse(BaseModel):
    id: int
    title: str
    slug: str
    description: str
    short_description: Optional[str] = None
    level: str
    estimated_hours: float
    instructor: InstructorSummary
    modules: List[ModuleWithLessons] = []
    is_enrolled: bool = False
    progress_percentage: float = 0.0
    current_lesson_id: Optional[int] = None

    class Config:
        from_attributes = True


class LessonDetailResponse(BaseModel):
    id: int
    title: str
    slug: str
    content: str
    lesson_type: str
    order: int
    duration_minutes: int
    is_completed: bool = False
    module_id: int
    module_title: str
    course_id: int
    course_title: str
    prev_lesson_id: Optional[int] = None
    next_lesson_id: Optional[int] = None

    class Config:
        from_attributes = True


class LessonCompleteResponse(BaseModel):
    lesson_id: int
    is_completed: bool
    course_id: int
    course_progress_percentage: float
    is_course_completed: bool
    points_awarded: int
    next_lesson_id: Optional[int] = None


class RecentActivityItem(BaseModel):
    id: str
    title: str
    activity_type: str  # lesson_completed, course_started, badge_unlocked
    timestamp: str
    description: str
    route: str


class ActiveCourseSummary(BaseModel):
    course_id: int
    course_title: str
    current_lesson_id: int
    current_lesson_title: str
    progress_percentage: float


class StudentProgressStats(BaseModel):
    overall_progress_percentage: float
    completed_lessons_count: int
    total_lessons_count: int
    in_progress_courses_count: int
    completed_courses_count: int
    total_learning_hours: float
    current_streak_days: int
    total_points: int
    today_goal_completed: bool
    active_course: Optional[ActiveCourseSummary] = None
    recommended_course: Optional[CourseListItem] = None
    recent_activities: List[RecentActivityItem] = []

File: frontend/src/types/course.ts
export interface LessonSummary {
  id: number;
  title: string;
  slug: string;
  lesson_type: string;
  order: number;
  duration_minutes: number;
  is_completed: boolean;
}

export interface ModuleWithLessons {
  id: number;
  title: string;
  description?: string;
  order: number;
  lessons: LessonSummary[];
}

export interface InstructorSummary {
  id: number;
  full_name: string;
  role: string;
  bio?: string;
  avatar_url?: string;
}

export interface CourseListItem {
  id: number;
  title: string;
  slug: string;
  short_description?: string;
  level: string;
  estimated_hours: number;
  modules_count: number;
  lessons_count: number;
  is_enrolled: boolean;
  progress_percentage: number;
}

export interface CourseDetailResponse {
  id: number;
  title: string;
  slug: string;
  description: string;
  short_description?: string;
  level: string;
  estimated_hours: number;
  instructor: InstructorSummary;
  modules: ModuleWithLessons[];
  is_enrolled: boolean;
  progress_percentage: number;
  current_lesson_id?: number;
}

export interface LessonDetailResponse {
  id: number;
  title: string;
  slug: string;
  content: string;
  lesson_type: string;
  order: number;
  duration_minutes: number;
  is_completed: boolean;
  module_id: number;
  module_title: string;
  course_id: number;
  course_title: string;
  prev_lesson_id?: number;
  next_lesson_id?: number;
}

export interface LessonCompleteResponse {
  lesson_id: number;
  is_completed: boolean;
  course_id: number;
  course_progress_percentage: number;
  is_course_completed: boolean;
  points_awarded: number;
  next_lesson_id?: number;
}

export interface RecentActivityItem {
  id: string;
  title: string;
  activity_type: string;
  timestamp: string;
  description: string;
  route: string;
}

export interface ActiveCourseSummary {
  course_id: number;
  course_title: string;
  current_lesson_id: number;
  current_lesson_title: string;
  progress_percentage: number;
}

export interface StudentProgressStats {
  overall_progress_percentage: number;
  completed_lessons_count: number;
  total_lessons_count: number;
  in_progress_courses_count: number;
  completed_courses_count: number;
  total_learning_hours: number;
  current_streak_days: number;
  total_points: number;
  today_goal_completed: boolean;
  active_course?: ActiveCourseSummary;
  recommended_course?: CourseListItem;
  recent_activities: RecentActivityItem[];
}
